src/sparsematrix.jl

```diff
@@ -209,8 +209,13 @@ nnz(S::LowerTriangular{<:Any,<:AbstractSparseMatrixCSC}) = nnz1(S)
 nnz(S::SparseMatrixCSCView) = nnz1(S)
 nnz1(S) = sum(length.(nzrange.(Ref(S), axes(S, 2))))
 
-function count(pred, S::AbstractSparseMatrixCSC)
-    count(pred, nzvalview(S)) + pred(zero(eltype(S)))*(prod(size(S)) - nnz(S))
+Base._count(f, A::Adjoint{<:Any,<:AbstractSparseMatrixCSC}, ::Colon, init) =
+    Base._simple_count(f∘adjoint, parent(A), init)
+Base._count(f, A::Transpose{<:Any,<:AbstractSparseMatrixCSC}, ::Colon, init) =
+    Base._simple_count(f∘transpose, parent(A), init)
+
+function Base._simple_count(pred, S::AbstractSparseMatrixCSC, init::T) where T
+    init + T(count(pred, nzvalview(S)) + pred(zero(eltype(S)))*(prod(size(S)) - nnz(S)))
```

Comment:
is there any reason only one is converted to T?

Reply:
Because `init` is already of type `T`, and I was assuming that `(T + T)::T`. But wait, I believe I read somewhere that the type of `init` determines the return type, but I can't find it anymore. I think this conversion non-sense.

Reply:
Well, I found it again: it's in the `count` docstring, so I better leave it.